src/qililab/result/experiment_results_writer.py

```diff
@@ -0,0 +1,250 @@
+
+                    # Attach dimension scales (loops) to the results dataset
+                    for idx, dim_variables in enumerate(qprogram_data["dims"]):
+                        for dim_variable in dim_variables:
+                            results_ds.dims[idx].attach_scale(qloop_group[dim_variable])
+                        results_ds.dims[idx].label = ",".join(list(dim_variables))
+                    for idx, dim_variables in enumerate(measurement_data["dims"]):
+                        for dim_variable in dim_variables:
+                            results_ds.dims[len(qprogram_data["dims"]) + idx].attach_scale(mloop_group[dim_variable])
+                        results_ds.dims[len(qprogram_data["dims"]) + idx].label = ",".join(list(dim_variables))
+
+                    # Attach the extra dimension (usually for I/Q) to the results dataset
+                    results_ds.dims[len(qprogram_data["dims"]) + len(measurement_data["dims"])].label = "I/Q"
+
+    def _create_resuts_access(self):
+        """Sets up internal data structures to allow for real-time data writing to the HDF5 file."""
+        if "qprograms" in self._metadata:
+            for qprogram_name, qprogram_data in self._metadata["qprograms"].items():
+                for measurement_name, _ in qprogram_data["measurements"].items():
+                    self.data[(qprogram_name, measurement_name)] = self._file[
+                        f"qprograms/{qprogram_name}/measurements/{measurement_name}/results"
+                    ]
+
+    def __enter__(self):
+        """Opens the HDF5 file and creates the structure for streaming.
+
+        Returns:
+            ExperimentResultsWriter: The ExperimentResultsWriter instance.
+        """
+        self._file = h5py.File(self.path, mode="w")
+        self._create_results_file()
+        self._create_resuts_access()
+
+        return self
+
+    def __setitem__(self, key: tuple, value: float):
+        """Sets an item in the results dataset.
+
+        Args:
+            key (tuple): A tuple of (qprogram_name or index, measurement_name or index, *indices).
+            value (float): The value to set at the specified indices.
+        """
+        qprogram_name, measurement_name, *indices = key
+        if isinstance(qprogram_name, int):
+            qprogram_name = f"QProgram_{qprogram_name}"
+        if isinstance(measurement_name, int):
+            measurement_name = f"Measurement_{measurement_name}"
+        self.data[(qprogram_name, measurement_name)][tuple(indices)] = value
+
+    @ExperimentResults.platform.setter
+    def platform(self, platform: str):
+        """Sets the YAML representation of the platform.
+
+        Args:
+            platform (str): The YAML string representing the platform.
+        """
+        path = ExperimentResults.PLATFORM_PATH
+        if path in self._file:
+            del self._file[path]
+        self._file[path] = platform
```

Comment:
I don't understand why are you deleting `self._file[path]`, isn't it overwritten in the next line?

I guess an h5py file does not get overwritten?

Reply:
If I don't delete it, I get the following error:

![image](https://github.com/user-attachments/assets/7625dc2e-82e9-4dd2-888d-60143440072f)